lib/groupBy.js

```diff
@@ -0,0 +1,37 @@
+import doLimit from './internal/doLimit';
+import groupByLimit from './groupByLimit';
+
+/**
+ * Returns a new object, where each value corresponds to an array of items, from
+ * `coll`, that returned the corresponding key. That is, the keys of the object
+ * correspond to the values passed to the `iteratee` callback. Note: Since this
+ * function applies the `iteratee` to each item in parallel, there is no
+ * guarantee that the `iteratee` functions will complete in order and there is
+ * no guarantee that grouped items will appear in the same order as in `coll`.
```

Comment:
Just to make sure I understand, do you mean kinda like `./internal/filter.js` where it asynchronously transforms the values, but synchronously filters them afterwards? I could look into that tomorrow.